notesapi/v1/tests/test_views.py

```diff
@@ -959,6 +959,26 @@ def test_pagination_page_start(self, page, start, previous_page, next_page):
             start=start
         )
 
+    @ddt.unpack
+    @ddt.data(
+        {"text": u"Ammar محمد عمار Muhammad", "search": u"محمد عمار", "tags": [u"عمار", u"Muhammad", u"محمد"]},
+        {"text": u"Ammar محمد عمار Muhammad", "search": u"محمد", "tags": [u"محمد", u"Muhammad"]},
+        {"text": u"Ammar محمد عمار Muhammad", "search": u"عمار", "tags": [ u"ammar", u"عمار"]},
+        {"text": u"Muhammad Ammar", "search": u"عمار", "tags": [ u"ammar", u"عمار"]},
+        {"text": u"محمد عمار", "search": u"Muhammad", "tags": [ u"Muhammad", u"عمار"]}
+    )
```

Comment:
The namedtuple solution mostly solves a similar problem to this dict solution.  I didn't actually know ddt would unpack dicts.  One advantage of namedtuples is that it provides stricter type checking on the ddt data.  Another is that you can hack the default values to reduce verbosity in some cases.  The disadvantage is that you have to declare your namedtuple, which is a bit redundant.

To see how we did it, check out https://github.com/edx/edx-platform/blob/master/lms/djangoapps/course_blocks/transformers/tests/test_user_partitions.py#L285